spec/support/transactions/operation.rb

```diff
@@ -100,27 +101,24 @@ def has_results?
       #
       # @return [ Result ] The result of executing the operation.
       #
-      # @since 2.0.0
+      # @since 2.6.0
       def execute(collection)
-        begin
-          if SESSION_OPERATIONS.include?(OPERATIONS[name])
-            send(OPERATIONS[name], options[:session])
-          else
-            send(OPERATIONS[name], collection)
-          end
-        rescue Mongo::Error::OperationFailure => e
-          { 'errorCodeName' => e.instance_variable_get(:result).send(:first_document)['codeName'] }
-        rescue Mongo::Error => e
-          { 'errorContains' => e.message }
+        if SESSION_OPERATIONS.include?(OPERATIONS[name])
+          send(OPERATIONS[name], options[:session])
+        else
+          send(OPERATIONS[name], collection)
         end
+      rescue Mongo::Error::OperationFailure => e
+        { 'errorCodeName' => e.instance_variable_get(:result).send(:first_document)['codeName'] }
+      rescue Mongo::Error => e
+        { 'errorContains' => e.message }
       end
 
       private
 
       def start_transaction(session)
-        opts = options
```

Comment:
I noticed that the `options` method actually creates a new hash each time it's called (through the `reduce` method, so copying the options (which is done incorrectly here) isn't even necessary in the first place